rest-service/manager_rest/resource_manager.py

```diff
@@ -417,27 +417,25 @@ def _get_queued_executions(self, deployment_storage_id):
             )
             .all()
         )
-
         # deployments we've already emitted an execution for - only emit 1
         # execution per deployment
         seen_deployments = set()
-        # {group: how many executions can it still run}
-        group_can_run = {}
+
         for execution in queued_executions:
             for group in execution.execution_groups:
-                if group not in group_can_run:
-                    group_can_run[group] = group.concurrency -\
-                        len(group.currently_running_executions())
+                if group._storage_id not in groups:
+                    groups[group._storage_id] = [0, group.concurrency]
 
-            if any(group_can_run[g] <= 0 for g in execution.execution_groups):
+            if any(groups[g._storage_id][0] >= groups[g._storage_id][1]
+                   for g in execution.execution_groups):
                 # this execution cannot run, because it would exceed one
                 # of its' groups concurrency limit
                 continue
             if execution._deployment_fk in seen_deployments:
                 continue
 
-            for group in execution.execution_groups:
-                group_can_run[group] -= 1
+            for g in execution.execution_groups:
+                groups[g._storage_id][0] += 1
```

Comment:
are you sure of `+`?

Reply:
yeah because before it used to be a "how many can the group still run" counter, but now it's a "how many is the group currently running" counter

and the check used to be something like `<= 0`, but now the check is `if any(groups[g._storage_id][0] >= groups[g._storage_id][1]`